manpages/man1/wazzuf-pre-ripper.1
.TH WAZZUF-PRE-RIPPER 1 "September 24, 2012"
.SH NAME
wazzuf-pre-ripper \- DVD/BD Analyser for wazzuf-ripper.
.SH DESCRIPTION
\fBwazzuf-pre-ripper\fP is a shell script (bash) to
.B  analyse
DVD/BD, helping providing
.BR wazzuf-ripper.conf (4)
for
.BR wazzuf-ripper (1)
.
.PP
It can search for IMdb ID and download poster image with imdb-cli tool.
.PP
It can also copy physical DVD to ISO file in order to ease ripping.
.PP
.SH SEE ALSO
.BR wazzuf-ripper (1)
,
.BR wazzuf-external-tools-install (1)
,
.BR wazzuf-ripper.conf (4)
,
.BR wazzuf-ripper-global.conf (4)
.
.SH AUTHOR
wazzuf-pre-ripper was written by Nicolas Perrin <booloki@lokizone.net>.
.PP
This manual page was written by Nicolas Perrin <booloki@lokizone.net>,
for the Debian project (and may be used by others).


changelog
# Wazzuf Ripper
# DVD/BD rip script
# booloki@gmail.com
# Changelog file

2012-09-12
* Add processor core numbers recognition

2012-09-10
* Lots of little improvements
* New wazzuf-path.conf file
* New IMDB_ID variable
* Add external tools installer (wazzuf-external-tools-install.sh)
* Add IMdb informations retrieve / poster image download (wazzuf-functions-imdb)
* Add .nfo maker (wazzuf-functions-nfo)

2012-09-09
* bugfix: Chapters file merge

2012-09-07
* DVD/ISO pre-rip script improvements
* New wazzuf-functions-pre file

2012-09-06
* Configuration file reorganization
* Add DVD_EPISODES_ORG and DVD_TITLE_LIST options
* Change EPISODE_FIRST/LAST to DVD_CHAPTER_FIRST/LAST options
* New wazzuf-functions-xmltag file
* New BASE_WORKING_FILE variable
* SHOW: Season & Episode denominations + types variables
* Chapters file improvements (DVD/ISO)

2012-09-05
* bugfix + improvements: cover art management
* Change LPCM to PCM
* Add DVD audio track DUMP option (no encoding) and removing DTS option
* bugfix: avoid DVD chapters informations merge when only one chapter

2012-09-04
* m2ts filename bugfix (pre-ripper)

2012-08-31
* bugfix: right subtitles path provided, and others small bugfixes

2012-08-29
* Add check_vobcopy and check_mkisofs
* Tagging bugfixes
* Add external srt subtitle Unicode check
* Add copy DVD to iso feature
* Some minor bugfixes / check
* Add tags to mp3 (ID3v2) and ogg files
* New video type management: MOVIE SHOW MUSIC instead of SERIE
* Add xml tags to mkv depending on video type (3 templates)

2012-08-28
* Add check_imagemagick function
* Add cover art management: resizing + thumbnail

2012-08-27
* Typo fixes
* External subtitle support improvements (SUBTITLE_X_FILE to SUBTITLE_X_FILE_FORCE)
* Add check_avconv check_7z check_tcdemux
* Add DVD structure display (pre-rip script)
* Add cover image MIME type recognition (jpg and png only)
* Add source files directory option for m2ts/ts, iso, external subtitles, cover image, episodes list...
=> lots of absolute to relative option changes in configuration file !
* Add audio sync option

2012-08-26
* Bugfix: apply SUBTITLE_NODEFAULT_FORCE option to second subtitle track

2012-07-10
* Bugfix: chapters informations re-taken into account

2012-06-14
* Add DVD video track DUMP option (no encoding)

2012-06-05
* Add optional mkv image attachment

2012-05-15
* Add physical DVD presence check

2012-05-14
* Pre-rip script improvements
* Add basic error catching

2012-05-07
* Bugfix: LCPM audio format re-taken into account

2012-04-23
* Set mencoder log level to 1 (Errors and Warning only) 

2012-04-19
* Wazzuf Ripper v2.0
* Add wazzuf-pre-ripper.sh script helping to fill configuration file

2012-04-18
* Add manual choice to bypass iso media source if not needed

2012-04-17
* Add autoaspect to mencoder xvid option to autodetect movie aspect

2012-04-09
* Add VIDEO_NOSKIP_FORCE option to force mencoder frames skipping

2012-04-08
* Add second subtitles track rip feature
* VIDEO_RATIO is now VIDEO_RATIO_FORCE option
* remove SUBTITLE_LANG check - mkvmerge do it well

2012-03-29
* New wazzuf-functions-check file
* Add second audio track rip feature
* Code structure change for v2
* New v1 release branch

2012-03-12
* Add SUBTITLE_NODEFAULT_FORCE option to avoid unnecessary default subtitle display
* Configuration file reorganization

2012-03-12
* Add licence.txt and readme.txt files
* Add ioniceness to audio extraction
* Remove useless NONE media support option
* Add DTS output support (remove AUDIO_DTS, add mandatory AUDIO_SOURCE)
* Video bitrate choice bugfix
* Add force multichapters option for one video (MULTICHAP_FORCE MULTICHAP_FIRST MULTICHAP_LAST)
* Change mkvmerge composition: add audio/subtitle track name (AUDIO_NAME SUBTITLE_NAME)
* Set default subtitle track to none
* Some configuration file improvements
* Add this changelog file to git

2012-03-05
* Add original id to working audio/subtitles filename to prepare multi-audio and multi-subtitles rip
* Separate video bitrate variables between DVD and BD
* Some minor bugfixes and comment changes
* Add CONF_FILE variable in script



wazzuf-pre-ripper.sh
#! /bin/bash
# Wazzuf Ripper
# DVD/BD pre-rip script helping to fill configuration file
# booloki@gmail.com

PATH_FILE="wazzuf-path.conf"
source $PATH_FILE

# check wazzuf files
source $FUNCTIONS_CHECK
checkandsource_wazzuf_files

check_mplayer
check_avconv

ISO_FILE_PATH=$SOURCE_DIRECTORY/$ISO_FILE
M2TS_FILE_PATH=$SOURCE_DIRECTORY/$M2TS_FILE

# enter source directory
mkdir -p $SOURCE_DIRECTORY
cd $SOURCE_DIRECTORY

# pre-rip source choice
case $SOURCE in
BD )
	if [ ! -f "$M2TS_FILE_PATH" ]
	then
		echo -ne "\n *************************************\n"
		echo " M2TS_FILE_PATH $M2TS_FILE_PATH does not exists ! Exiting..."
		echo -ne " *************************************\n"
		exit 1
	fi

	echo -ne " *************************************\n"
	echo " MPlayer informations"
	mplayer -v -vo null -ao null -frames 0 -identify "$M2TS_FILE_PATH" 2>/dev/null > M2TS-mplayer.info
	echo "-> see M2TS-mplayer.info"

	echo -ne " *************************************\n"
	echo " avconv informations"
	avconv -i "$M2TS_FILE_PATH" 2> M2TS-avconv.info
	echo "-> see M2TS-avconv.info"

	echo -ne " *************************************\n"
	echo M2TS_FILE_PATH=\"$M2TS_FILE_PATH\"
	cropdetect $M2TS_FILE_PATH

	echo -ne "*************************************\n"
	echo "All tracks:"
	avconv-decode M2TS-avconv.info
	;;
DVD )
	DVD_PATH="/dev/dvd"
	check_lsdvd
	echo -ne "*************************************\n"
	echo " lsdvd informations"
 	lsdvd -acsv $DVD_PATH 2>/dev/null > DVD-lsdvd.info
	echo "-> see DVD-lsdvd.info"

	echo -ne "*************************************\n"
	echo " MPlayer informations"
	mplayer -v -vo null -ao null -frames 0 -identify $DVD_PATH 2>/dev/null > DVD-mplayer.info
	echo "-> see DVD-player.info"

	echo -ne "*************************************\n"
	echo " avconv informations"
	avconv -i $DVD_PATH 2> DVD-avconv.info
	echo "-> see ISO-avconv.info"

	check_7z
	echo -ne "*************************************\n"
	echo " DVD structure informations"
	7z l $DVD_PATH > DVD-DVDstruct.info
	echo "-> see ISO-DVDstruct.info"

	echo -ne "*************************************\n"
	grep "^Disc Title" DVD-lsdvd.info
	DVD_TITLE_NUMBER=`grep "^Longest track:" DVD-lsdvd.info | sed s/'Longest track: '//`
	if  [[ $VIDEO_TYPE == "SHOW" && $DVD_EPISODES_ORG == "TITLES" ]]; then
		for DVD_TITLE_NUMBER in $DVD_TITLE_LIST
		do
			echo "######### Title: $DVD_TITLE_NUMBER #########"
			lsdvd -t $DVD_TITLE_NUMBER -acsv $DVD_PATH 2>/dev/null | grep -E "Subtitle|Audio|VTS" > DVD-lsdvd-T$DVD_TITLE_NUMBER.info
			echo `grep "Title: " DVD-lsdvd-T$DVD_TITLE_NUMBER.info | cut -d "," -f 2`
			cropdetect "dvd://$DVD_TITLE_NUMBER"
			grep VTS DVD-lsdvd-T$DVD_TITLE_NUMBER.info
			lsdvd-decode DVD-lsdvd-T$DVD_TITLE_NUMBER.info
		done
	else
		echo "Longest track chosen for crop detection: Title $DVD_TITLE_NUMBER"
		echo `grep "Title: $DVD_TITLE_NUMBER" DVD-lsdvd.info | cut -d "," -f 2`
		cropdetect "dvd://$DVD_TITLE_NUMBER"

		echo -ne "*************************************\n"
		echo "Title $DVD_TITLE_NUMBER Tracks:"
		lsdvd -t $DVD_TITLE_NUMBER -acsv $DVD_PATH 2>/dev/null | grep -E "Subtitle|Audio|VTS" > DVD-lsdvd-T$DVD_TITLE_NUMBER.info
		grep VTS DVD-lsdvd-T$DVD_TITLE_NUMBER.info
		lsdvd-decode DVD-lsdvd-T$DVD_TITLE_NUMBER.info

		echo "Note: For multiple Title crop detection, set VIDEO_TYPE to SHOW and DVD_EPISODES_ORG to TITLES"
	fi

	# choice to copy DVD to iso file
	DVD_MOUNT_PATH=`mount | grep udf | cut -d ' ' -f 3`
	DVD_MOUNT_NAME=`echo $DVD_MOUNT_PATH | cut -d '/' -f 3`
	echo -ne "*************************************\n"
	echo -ne " Wanna copy $DVD_MOUNT_PATH to $SOURCE_DIRECTORY/$ISO_FILE file ? (N/y)\n"
	read answer
	case $answer in
	y* | Y* )
		copy-dvd2iso
		echo -ne "\n You can now use ISO instead of DVD in configuration file."
		;;
	* )
		echo -ne "Skip DVD copy.\n"
		;;
	esac
	;;
ISO )
	if [ ! -f "$ISO_FILE_PATH" ]
	then
		echo -ne "\n *************************************\n"
		echo " ISO_FILE_PATH $ISO_FILE_PATH does not exists ! Exiting..."
		echo -ne " *************************************\n"
		exit 1
	fi

	check_lsdvd
	echo -ne "*************************************\n"
	echo " lsdvd informations"
	lsdvd -acsv "$ISO_FILE_PATH" 2>/dev/null > ISO-lsdvd.info
	echo "-> see ISO-lsdvd.info"
	
	echo -ne "*************************************\n"
	echo " MPlayer informations"
	mplayer -v -vo null -ao null -frames 0 -identify "$ISO_FILE_PATH" 2>/dev/null > ISO-mplayer.info
	echo "-> see ISO-mplayer.info"

	echo -ne "*************************************\n"
	echo " avconv informations"
	avconv -i "$ISO_FILE_PATH" 2> ISO-avconv.info
	echo "-> see ISO-avconv.info"

	check_7z
	echo -ne "*************************************\n"
	echo " DVD structure informations"
	7z l "$ISO_FILE_PATH" > ISO-DVDstruct.info
	echo "-> see ISO-DVDstruct.info"

	echo -ne "*************************************\n"
	echo "Tested iso:\"$ISO_FILE_PATH\""
	grep "^Disc Title" ISO-lsdvd.info
	DVD_TITLE_NUMBER=`grep "^Longest track:" ISO-lsdvd.info | sed s/'Longest track: '//`
	if  [[ $VIDEO_TYPE == "SHOW" && $DVD_EPISODES_ORG == "TITLES" ]]; then
		for DVD_TITLE_NUMBER in $DVD_TITLE_LIST
		do
			echo "######### Title: $DVD_TITLE_NUMBER #########"
			lsdvd -t $DVD_TITLE_NUMBER -acsv "$ISO_FILE_PATH" 2>/dev/null | grep -E "Subtitle|Audio|VTS" > ISO-lsdvd-T$DVD_TITLE_NUMBER.info
			echo `grep "Title: " ISO-lsdvd-T$DVD_TITLE_NUMBER.info | cut -d "," -f 2`
			cropdetect "-dvd-device $ISO_FILE_PATH dvd://$DVD_TITLE_NUMBER"
			grep VTS ISO-lsdvd-T$DVD_TITLE_NUMBER.info
			lsdvd-decode ISO-lsdvd-T$DVD_TITLE_NUMBER.info
		done
	else
		echo "Longest track chosen for crop detection: Title $DVD_TITLE_NUMBER"
		echo `grep "Title: $DVD_TITLE_NUMBER" ISO-lsdvd.info | cut -d "," -f 2`
		cropdetect "-dvd-device $ISO_FILE_PATH dvd://$DVD_TITLE_NUMBER"

		echo -ne "*************************************\n"
		echo "Title $DVD_TITLE_NUMBER Tracks:"
		lsdvd -t $DVD_TITLE_NUMBER -acsv "$ISO_FILE_PATH" 2>/dev/null | grep -E "Subtitle|Audio|VTS" > ISO-lsdvd-T$DVD_TITLE_NUMBER.info
		grep VTS ISO-lsdvd-T$DVD_TITLE_NUMBER.info
		lsdvd-decode ISO-lsdvd-T$DVD_TITLE_NUMBER.info

		# not so interesting
		#avconv-decode ISO-avconv.info

		echo "Note: For multiple Title crop detection, set VIDEO_TYPE to SHOW and DVD_EPISODES_ORG to TITLES"
	fi
	;;
* )
        echo -ne "\n Media source problem: make sure you filled DVD, ISO or BD in configuration file\n"
        exit 1
        ;;
esac

# IMDb: get ID and download poster image
echo -ne "*************************************\n"
echo -ne "\n Try to get IMDb ID with $TITLE_NAME ($DATE)  ? (Y/n)\n"
read IMDB_ANSWER
case $IMDB_ANSWER in
N* | n* )
	echo -ne " Skip IMDb informations.\n"
	;;
* )
	# get response
	get_imdb_response_title
	if  [[ $IMDB_RESPONSE == "True" ]]; then
		get_imdb_id
		echo -ne " IMdb ID seems to be $IMDB_ID\n"
		echo -ne " -> Please check in http://www.imdb.com/title/$IMDB_ID\n"
		echo -ne "*************************************\n"

		# download cover or not
		echo -ne "\n*************************************\n"
		echo -ne " Download $TITLE_NAME ($DATE) IMDb cover ? (N/y)\n"
		read IMDB_ANSWER
		case $IMDB_ANSWER in
		Y* | y* )
			download_imdb_poster
			;;
		* )
			echo -ne " Skip IMDb cover download.\n"
			;;
		esac
		echo -ne "*************************************\n"
	else
		# $IMDB_RESPONSE == "False"
		echo -ne " No IMdb informations found ! Try with another release date (first season release date for show)\n"
		echo -ne " Or directly search at http://www.imdb.com -> IMdb ID is XXXXXXXXX in URL: http://www.imdb.com/title/XXXXXXXXX \n"
	fi
	;;
esac


exit 0